### ISSUE TITLE
EKS cluster AWSMachines not created because MachineSet pre-fight check fails 

### ISSUE THREAD

**Issue by tingchienc** [kubernetes-sigs/cluster-api-provider-aws#5225]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

Add a MachineDeployment for the EKS cluster deployed by CAPA
No errors in the capa-controller pods

But there is an error for the machineDeployment.

Error messages:
Message:               Performing "Scale up" on hold because ControlPlane namespace/cluster-control-plane is provisioning ("ControlPlaneIsStable" preflight failed). The operation will continue after the preflight check(s) pass

After we set the annotation "machineset.cluster.x-k8s.io/skip-preflight-checks: ControlPlaneIsStable", the EC2 machines finally created.

The problem is why the preflight-checks fail? We don't want to add the annotation for every MachineDeployment.
And it seems like a new feature added in Sep by ClusterAPI.

**What did you expect to happen:**
EC2 machines created without adding the annotation

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: latest
- Kubernetes version: (use `kubectl version`): 1.29.0
- OS (e.g. from `/etc/os-release`): ubuntu

**Comment by oneironautjack** [comment]
Experiencing the same issue but using existing VPC without EKS. Did you find a permanent solution or are you still using the workaround?

**Comment by mfrehe** [comment]
Hello,

i am facing the exact same issue.

Cluster-api-provider-aws version: v2.7.1
cluster-api version: v1.9.3
Kubernetes version: (use kubectl version): 1.30.0

I think it is related to the fact that the **MachineSetPreflightCheck** was introduced as an experimental feature in version 1.5 and has been promoted to beta status in version 1.9, where it is enabled by default.

https://github.com/kubernetes-sigs/cluster-api/issues/11117

Not sure, but maybe the check for the controlplane version is not working for all controlplanes.

https://github.com/kubernetes-sigs/cluster-api/blob/07d97c61f79cf920d6e0f3d9f5684677c2ec85de/internal/contract/controlplane.go#L192

At least i can just find a version `v1.30` in `.spec.version`, while `.status.version` does not exist at the controlplane. (But that's just a wild guess)

Regards
Marcel

**Comment by mfrehe** [comment]
Hello,

i am facing the same issue with EKS.

After some research i recognized that the `MachineSetPreflightCheck` was introduced as an experimental feature in CAPI version 1.5 and has been promoted to beta status in CAPI version 1.9, where it is enabled by default (https://github.com/kubernetes-sigs/cluster-api/issues/11117). 

So i'd assume that you are using CAPI >= v1.9.0 (like me) which seems to be not supported by CAPA v2.7.1

An issue for CAPI v1.9.0 was created in Nov 24: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/5218

As i am afraid of downgrading to CAPI v1.8.x (which version seems to be supported by CAPA, https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/5061), i'd like to ask for a CAPA version that supports CAPI >= v1.9.0.

Regards
Marcel

**Comment by oneironautjack** [comment]
Any luck with the downgrades? I've attempted to downgrade CAPI to v1.8.6 to no avail.

**Comment by damdo** [comment]
Thanks for reporting this.
We had a discussion about this on Slack this morning with the Core CAPI maintainers.
See it here: https://kubernetes.slack.com/archives/CD6U2V71N/p1739783013734149

TLDR: we are planning to update the status field for `AWSManagedControlPlane` to include `Version`, and make the CAPI Pre Flight checks happy.
I am working on it here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/5209

**Comment by damdo** [comment]
This was fixed
